tests/TestCase/Network/ResponseTest.php

```diff
@@ -2778,6 +2778,45 @@ public function testWithBody()
     }
 
     /**
+     * Test with string body.
+     *
+     * @expectedException \PHPUnit\Framework\Error\Error
+     * @expectedExceptionMessage Array to string conversion
+     * @return void
+     */
+    public function testWithStringBody()
+    {
+        $response = new Response();
+        $response->withStringBody('Foo');
+        $body = $response->getBody();
+        $this->assertSame('Foo', (string)$body);
+
+        $response = new Response();
+        $response->withStringBody('');
+        $body = $response->getBody();
+        $this->assertSame('', (string)$body);
+
+        $response = new Response();
+        $response->withStringBody(null);
+        $body = $response->getBody();
+        $this->assertSame('', (string)$body);
+
+        $response = new Response();
+        $response->withStringBody(1337);
+        $body = $response->getBody();
+        $this->assertSame('1337', (string)$body);
+
+        /*
+         * This should produce an "Array to string conversion" error
+         * which gets thrown as a \PHPUnit\Framework\Error\Error Exception by PHPUnit.
+         */
+        $response = new Response();
+        $response->withStringBody(['foo' => 'bar']);
+        $body = $response->getBody();
+        $this->assertSame('', (string)$body);
```

Comment:
You could make this a separate case. Right now the expected exception could mask failure elsewhere in the test.